cpp/src/io/parquet/writer_impl.cu

```diff
@@ -454,6 +456,60 @@ std::vector<schema_tree_node> construct_schema_tree(LinkedColVector const& linke
         add_schema(col->children[lists_column_view::child_column_index],
                    col_meta.child(lists_column_view::child_column_index),
                    schema.size() - 1);
+      } else if (col->type().id() == type_id::LIST && col_meta.is_map()) {
+        // Map schema is denoted by a list of struct
+        // e.g. List<Struct<String,String>> will be
+        // "col_name" : { "key_value" : { "key", "value" } }
+
+        // verify the List child structure is a struct<left_child, right_child>
+        auto const& struct_col = col->child(lists_column_view::child_column_index);
+        CUDF_EXPECTS(struct_col.type().id() == type_id::STRUCT, "Map should be a List of struct");
+        CUDF_EXPECTS(struct_col.num_children() == 2,
+                     "Map should be a List of struct with two children only but found " +
+                       std::to_string(struct_col.num_children()));
+
+        schema_tree_node map_schema{};
+        map_schema.converted_type = ConvertedType::MAP;
+        map_schema.repetition_type =
+          col_nullable ? FieldRepetitionType::OPTIONAL : FieldRepetitionType::REQUIRED;
+        map_schema.name         = col_meta.get_name();
+        map_schema.num_children = 1;
+        map_schema.parent_idx   = parent_idx;
+        schema.push_back(std::move(map_schema));
+
+        schema_tree_node repeat_group{};
+        repeat_group.repetition_type = FieldRepetitionType::REPEATED;
+        repeat_group.name            = "key_value";
+        repeat_group.num_children    = 2;
+        repeat_group.parent_idx      = schema.size() - 1;  // Parent is map_schema, last added.
+        schema.push_back(std::move(repeat_group));
+
+        CUDF_EXPECTS(col_meta.num_children() == 2,
+                     "List column's metadata should have exactly two children");
```

Comment:
One more, otherwise the subsequent lines would segfault.
```cpp
CUDF_EXPECTS(col_meta.child(lists_column_view::child_column_index).num_children() == 2,
                     "Map struct column's metadata should have exactly two children");
```